components/support/migration/src/main/java/mozilla/components/support/migration/FennecMigrator.kt

```diff
@@ -384,4 +412,18 @@ class FennecMigrator private constructor(
             }
         }
     }
+
+    @SuppressWarnings("TooGenericExceptionCaught")
+    private fun migrateGecko(): Result<Unit> {
+        if (profile == null) {
+            crashReporter.submitCaughtException(IllegalStateException("Missing Profile path"))
+            return Result.Failure(IllegalStateException("Missing Profile path"))
+        }
+
+        return try {
+            GeckoMigration.migrate(profile.path)
+        } catch (e: Exception) {
+            Result.Failure(e)
```

Comment:
`crashReporter.submitCaughtException` here as well?

Reply:
`migrateHistory`, `migrateBookmarks` and `migrateOpenTabs` don't do this for the generic exception case. Should they?

Reply:
The code that runs the migrations logs all failures. Maybe we should report all failure exceptions from that code?